File: 27.移除元素.cpp
/*
 * @lc app=leetcode.cn id=27 lang=cpp
 *
 * [27] 移除元素
 */

// @lc code=start
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int n = nums.size();
        int i = 0;
        while(i < n){
            if(nums[i] == val)
            {
                nums[i] = nums[n-1];
                n--;
            }
            else{
                i++;
            }
        }
        return i;
    }
};
// @lc code=end

File: pyround1/27.移除元素.py
#
# @lc app=leetcode.cn id=27 lang=python
#
# [27] 移除元素
#

# @lc code=start
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        for index in range(len(nums)-1, -1, -1):
            if nums[index] == val:
                nums.pop(index)
        return len(nums)
# @lc code=end

